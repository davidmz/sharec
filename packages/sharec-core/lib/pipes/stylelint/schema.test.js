const { fixtures } = require('testUtils')
const { stylelintJson } = require('./schema')

describe('pipes > stylelint > schema', () => {
  describe('JSON', () => {
    const stylelintFxt = fixtures('stylelint/json/00-base', 'map')

    it('should merge configs', () => {
      expect(stylelintJson(stylelintFxt)).toMatchSnapshot()
    })
  })
})
