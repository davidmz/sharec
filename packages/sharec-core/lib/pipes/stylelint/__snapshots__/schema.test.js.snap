// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipes > stylelint > schema JSON should merge configs 1`] = `
Map {
  "extends" => Array [
    "stylelint-config-standard",
    "stylelint-config-recess-order",
  ],
  "rules" => Map {
    "declaration-block-trailing-semicolon" => null,
    "block-closing-brace-newline-before" => null,
    "block-opening-brace-space-before" => null,
    "at-rule-no-unknown" => Array [
      false,
      Map {
        "ignoreAtRules" => Array [
          "define-mixin",
          "mixin",
        ],
      },
    ],
    "color-named" => "never",
  },
  "processors" => Array [
    Array [
      "stylelint-my-processor",
      Map {
        "optionOne" => false,
      },
    ],
    Array [
      "some-other-processor",
      Map {
        "optionOne" => false,
        "optionTwo" => true,
      },
    ],
    "another-processor",
  ],
}
`;
